lib/isolator.rb

```diff
@@ -69,27 +69,33 @@ def transactions_threshold
     end
 
     def transactions_threshold=(val)
+      # Do we need separate threshold per connection?
```

Comment:
Yeah, ideally we need to track threshold separately.

Currently, we only update this value when a DatabaseCleaner or Rails transactional tests are used.
The former doesn't seem to support multiple databases, thus, we need to increase the threshold only for the primary one.
The latter opens a transaction for every configuration but there could be dynamic configurations in the application code—thus, we should take care of them as well.

Reply:
For backward compatibility, we can default the primary connection's threshold to existing `isolator_threshold` and 1 for non-primary connections. However, if users want to fine tune the thresholds then they'd need the connection ids. Can you share how you envision the dynamic config api should work? How do we deal with dropped and re-established connection under a different id?